app/src/main/java/org/mozilla/fenix/settings/account/SyncController.kt

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.account
+
+import android.content.DialogInterface
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.provider.Settings
+import android.text.SpannableString
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.app.AlertDialog
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.settings.SupportUtils
+
+/**
+ * Controller for handling [SyncInteractor] requests.
+ */
+class SyncController(
```

Comment:
Let's have this be a `DefaultSyncController` that implements a `SyncController` interface to follow arch patterns elsewhere